File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cs
using System;

public class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        string s = "";
            
        while(n != 0)
        {
            s = n % k + s;
            n /= k;
        }
        string[] arr = s.Split("0");
        
        for(int i = 0; i < arr.Length; i++)
        {
            if(arr[i].Equals("")) continue;
            long num = long.Parse(arr[i]);
            if(IsPrime(num))
            {
                answer++;
            }
        }
        return answer;
    }
    
    private bool IsPrime(long n)
    {
        if(n == 1) return false;
        if(n == 2) return true;
        if(n % 2 == 0) return false;
        
        long d = (long)Math.Sqrt(n);
        for(long i = 3; i <= d; i+=2)
        {
            if(n % i == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.cpp
#include <string>
#include <vector>

using namespace std;

bool isPrime(long long n)
{
    if(n == 1) return false;
    if(n == 2) return true;
    
    for(long long i = 3; i * i <= n; i+= 2)
    {
        if(n % i == 0) return false;
    }
    return true;
}
string convert(int n, int k)
{
    string s = "";
    
    while(n > 0)
    {
        s = to_string(n % k) + s;
        n /= k;
    }
    return s;
}

int solution(int n, int k) {
    int answer = 0;
    
    string s = convert(n, k);
    string temp = "";
    
    for(char c : s)
    {
        if(c == '0')
        {
            if(temp != "")
            {
                long long num = stoll(temp);
                if(isPrime(num)) answer++;
                temp = "";
            }
        }
        else
            temp += c;
    }
    
    if(temp != "")
    {
        long long num = stoll(temp);
        if(isPrime(num)) answer++;
    }
    return answer;
}